src/main/java/com/nike/cerberus/util/AwsIamRoleArnParser.java

```diff
@@ -19,47 +19,98 @@
+import org.apache.commons.lang3.StringUtils;
 
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 /**
  * Utility class for concatenating and parsing AWS IAM role ARNs.
  */
-// TODO: remove
 public class AwsIamRoleArnParser {
 
     public static final String AWS_IAM_ROLE_ARN_TEMPLATE = "arn:aws:iam::%s:role/%s";
 
-    public static final String AWS_IAM_ROLE_ARN_REGEX = "^arn:aws:iam::(?<accountId>\\d+?):role/(?<roleName>.+)$";
+    public static final String AWS_IAM_PRINCIPAL_ARN_REGEX = "^arn:aws:(iam|sts)::(?<accountId>\\d+?):(?!group).+?/(?<roleName>.+)$";
 
-    public static final String AWS_IAM_PRINCIPAL_ARN_REGEX = "^arn:aws:(iam|sts)::.+$";
+    private static final String AWS_IAM_ROLE_ARN_REGEX = "^arn:aws:iam::(?<accountId>\\d+?):role/(?<roleName>.+)$";
+
+    private static final String AWS_IAM_ASSUMED_ROLE_ARN_REGEX = "^arn:aws:sts::(?<accountId>\\d+?):assumed-role/(?<roleName>.+)/.+$";
+
+    private static final String GENERIC_ASSUMED_ROLE_REGEX = "^arn:aws:sts::(?<accountId>\\d+?):assumed-role/.+$";
+
+    private static final Pattern IAM_PRINCIPAL_ARN_PATTERN = Pattern.compile(AWS_IAM_PRINCIPAL_ARN_REGEX);
 
     private static final Pattern IAM_ROLE_ARN_PATTERN = Pattern.compile(AWS_IAM_ROLE_ARN_REGEX);
 
-    public String getAccountId(String roleArn) {
+    private static final Pattern IAM_ASSUMED_ROLE_ARN_PATTERN = Pattern.compile(AWS_IAM_ASSUMED_ROLE_ARN_REGEX);
 
-        Matcher iamRoleArnMatcher = IAM_ROLE_ARN_PATTERN.matcher(roleArn);
+    private static final Pattern GENERIC_ASSUMED_ROLE_PATTERN = Pattern.compile(GENERIC_ASSUMED_ROLE_REGEX);
 
-        if (! iamRoleArnMatcher.find()) {
-            throw ApiException.newBuilder()
-                    .withApiErrors(new InvalidIamRoleArnApiError(roleArn))
-                    .build();
-        }
+    /**
+     * Gets account ID from a 'role' ARN
+     * @param roleArn - Role ARN to parse
+     * @return - Account ID
+     */
+    public String getAccountId(final String roleArn) {
+
+        return getNamedGroupFromPattern(IAM_ROLE_ARN_PATTERN, "accountId", roleArn);
+    }
+
+    /**
+     * Gets role name form a 'role' ARN
+     * @param roleArn - Role ARN to parse
+     * @return
+     */
+    public String getRoleName(final String roleArn) {
+
+        return getNamedGroupFromPattern(IAM_ROLE_ARN_PATTERN, "roleName", roleArn);
 
-        return iamRoleArnMatcher.group("accountId");
     }
 
-    public String getRoleName(String roleArn) {
+    /**
+     * Returns true if the ARN is in format 'arn:aws:iam::000000000:role/example' and false if not
+     * @param arn - ARN to test
+     * @return - True if is 'role' ARN, False if not
+     */
+    public boolean isRoleArn(final String arn) {
+
+        final Matcher iamRoleArnMatcher = IAM_ROLE_ARN_PATTERN.matcher(arn);
+
+        return iamRoleArnMatcher.find();
+    }
+
+    /**
+     * Converts a principal ARN (e.g. 'arn:aws:iam::0000000000:instance-profile/example') to a 'role' ARN,
```

Comment:
I see from your other comments how this doc can be confusing. I will find a way to rephrase.

However, to answer your question: no. A role in AWS is a container for IAM policies, while a principal is an actor who "assumes" that role and all of its permissions